src/EditorFeatures/TestUtilities/LanguageServer/AbstractLanguageServerProtocolTests.cs

```diff
@@ -474,11 +481,60 @@ internal TestLspServer(TestWorkspace testWorkspace, WellKnownLspServerKinds serv
                 GetManagerAccessor().ResetLspSolutions();
             }
 
-            public Task<ResponseType?> ExecuteRequestAsync<RequestType, ResponseType>(string methodName, RequestType request, LSP.ClientCapabilities clientCapabilities,
-                string? clientName, CancellationToken cancellationToken) where RequestType : class
+            private static JsonMessageFormatter CreateJsonMessageFormatter()
             {
-                return _requestDispatcher.ExecuteRequestAsync<RequestType, ResponseType>(
-                    _executionQueue, methodName, request, clientCapabilities, clientName, cancellationToken);
+                var messageFormatter = new JsonMessageFormatter();
+                LSP.VSInternalExtensionUtilities.AddVSInternalExtensionConverters(messageFormatter.JsonSerializer);
+                return messageFormatter;
+            }
+
+            internal static async Task<TestLspServer> CreateAsync(TestWorkspace testWorkspace, LSP.ClientCapabilities clientCapabilities, WellKnownLspServerKinds serverKind)
```

Comment:
we now directly create the jsonrpc stream and LanguageServerTarget.  Tests will now go through serialization / deserialization as well as the full functionality of the LanguageServerTarget (yay more test coverage)

Reply:
This seems *really* good. :smile: